sparse_strips/vello_hybrid/src/schedule.rs

```diff
@@ -0,0 +1,230 @@
+    rounds: VecDeque<Round>,
+}
+
+/// A "round" is a coarse scheduling quantum.
+///
+/// It represents draws in up to three render targets; two for intermediate
+/// clip/blend buffers, and the third for the actual render target. The two
+/// clip buffers are for even and odd clip depths.
+#[derive(Default)]
+struct Round {
+    draws: [Draw; 3],
+    /// Slots that will be freed after the draws
+    free: [Vec<usize>; 2],
+}
+
+/// State for a single tile.
+///
+/// Perhaps this should just be a field in the scheduler.
+#[derive(Default)]
+struct TileState {
+    stack: Vec<TileEl>,
+}
+
+#[derive(Clone, Copy)]
+struct TileEl {
+    slot_ix: usize,
+    round: usize,
+}
+
+#[derive(Default)]
+struct Draw(Vec<GpuStrip>);
+
+impl Schedule {
+    pub(crate) fn new(n_slots: usize) -> Self {
+        let free0: Vec<_> = (0..n_slots).collect();
+        let free1 = free0.clone();
+        let free = [free0, free1];
+        let mut rounds = VecDeque::new();
+        rounds.push_back(Round::default());
+        Self {
+            round: 0,
+            free,
+            rounds,
+        }
+    }
+
+    pub(crate) fn do_scene(&mut self, junk: &mut RendererJunk<'_>, scene: &Scene) {
+        let mut state = TileState::default();
+        let wide_tiles_per_row = (scene.width).div_ceil(WideTile::WIDTH);
+        let wide_tiles_per_col = (scene.height).div_ceil(Tile::HEIGHT);
+        for wide_tile_row in 0..wide_tiles_per_col {
+            for wide_tile_col in 0..wide_tiles_per_row {
+                let wide_tile_idx = usize::from(wide_tile_row) * usize::from(wide_tiles_per_row)
+                    + usize::from(wide_tile_col);
+                let wide_tile = &scene.wide.tiles[wide_tile_idx];
+                let wide_tile_x = wide_tile_col * WideTile::WIDTH;
+                let wide_tile_y = wide_tile_row * Tile::HEIGHT;
+                self.do_tile(junk, wide_tile_x, wide_tile_y, wide_tile, &mut state);
+            }
+        }
+        while !self.rounds.is_empty() {
+            self.flush(junk);
+        }
+    }
+
+    /// Flush one round.
+    ///
+    /// The rounds queue must not be empty.
+    fn flush(&mut self, junk: &mut RendererJunk<'_>) {
+        let round = self.rounds.pop_front().unwrap();
+        for (i, draw) in round.draws.iter().enumerate() {
+            if !draw.0.is_empty() {
+                junk.do_render_pass(&draw.0, self.round, i);
+            }
+        }
+        for i in 0..1 {
+            self.free[i].extend(&round.free[i]);
+        }
+        self.round += 1;
+    }
+
+    #[allow(clippy::todo, reason = "still working on this")]
+    fn do_tile(
+        &mut self,
+        junk: &mut RendererJunk<'_>,
+        wide_tile_x: u16,
+        wide_tile_y: u16,
+        tile: &WideTile,
+        state: &mut TileState,
+    ) {
+        state.stack.clear();
+        state.stack.push(TileEl {
+            slot_ix: !0,
+            round: self.round,
+        });
+        let bg = tile.bg.to_u32();
```

Comment:
Is `bg` going to be used outside of this alpha check and then the conditionalized draw?